app/routines/export_users_last_login_date.rb

```diff
@@ -0,0 +1,52 @@
+class ExportUsersLastLoginDate
+  lev_routine
+
+  protected
+
+  def exec(delete_after: true)
+    outputs[:filename] = filename
+    outputs[:info] = info
+    generate_csv
+    remove_exported_files if delete_after
+  end
+
+  def info
+    output_users = []
+    User.preload(:contact_infos).find_each do |user|
+      output_users << Hashie::Mash.new({
+        emails: user.contact_infos.map(&:value).join(", "),
+        last_login_at: SecurityLog.sign_in_successful
+                                    .where(user_id: user.id).try(:last).try(:created_at).try(:strftime, "%m/%d/%Y %I:%M%p %Z")
+      })
+    end
+    output_users
+  end
+
+  def generate_csv
+    CSV.open(filename, 'w') do |file|
+      file.add_row ([
+          "Email(s)",
+          "Last login date",
+      ])
+
+      info.each do |hashie|
+        file.add_row([
+          hashie.emails,
+          hashie.last_login_at,
+        ])
+      end
+    end
+  end
+
+  def filename
+    File.join exports_folder, "users_last_login_date.csv"
+  end
+
+  def exports_folder
+    File.join 'tmp'
```

Comment:
`join` with 1 argument is odd. Maybe just `tmp`.

Reply:
Copy & pasta 🍝 . Fixed.